### DISCUSSION TITLE
Managing External State: dynamic authorization

### DISCUSSION THREAD

**Discussion by demamoh** [openfga/community#198] [Q&A]
In the context of doing dynamic authorization with large data sizes and from multiple sources of truth for data:

Is there documentation/best practices on state management?

I was hoping for a guide something like this: https://www.openpolicyagent.org/docs/latest/external-data/

An event-driven way I have seen this handled in OPA is with this tool: https://github.com/permitio/opal

What would be the best analog in OpenFGA?

**Comment by aaguiarz** [comment]
Hi @demamoh 

We don't have that documentation yet. In terms of the OPA document, you'd use the "Push Data" approach. 

There are no connectors for third-party data sources yet. 

What are the multiple sources of data you are thinking of?
